### PR TITLE
Support RPG Maker 2003 battle commands in RPG Maker 2000 battles

### PR DESCRIPTION
**Pull request by rueter37**
This PR enables the usage of the RPG Maker 2003 battle command feature in RPG Maker 2003 games, even if the RPG Maker 2000 battle system feature is enabled.

### CODE REVIEW

**Review by Ghabry — APPROVED** [review]
Thats fine for me. fmatthew also wanted that the battle systems share less code with each other because the flow is hard to follow so copy-paste is acceptable here.

Only ask you to append "2k3" to the following functions to make it more clear that they are enhancements: GetBattleCommandNames, SubskillSelected, SpecialSelected

**Review by fdelapena — APPROVED** [review]

**Comment by rueter37** [comment]
Did the requested function name changes now.

**Review by Ghabry — APPROVED** [review]
(when the other PR is in)

**Comment by rueter37** [comment]
Fixed.

**Comment by Ghabry** [comment]
Jenkins: Test this please

**Comment by Ghabry** [comment]
Another error guess this will go for a while enabled now the pr builder: 


game_player.cpp:637:69: error: no matching function for call to 'max(int, long int)'
  637 |                                                 int value = std::max(1, std::abs(hero->GetMaxHp() * terrain->damage / 100));

**Comment by rueter37** [comment]
Fixed.

**Comment by Ghabry** [comment]
All builds passed, guess some event did not go through
